src/js/tabs/balance.js

```diff
@@ -15,20 +14,22 @@ BalanceTab.prototype.mainMenu = 'wallet';
 
 BalanceTab.prototype.angularDeps = Tab.prototype.angularDeps.concat(['qr']);
 
-BalanceTab.prototype.generateHtml = function ()
-{
+BalanceTab.prototype.generateHtml = function() {
   return require('../../templates/tabs/balance.jade')();
 };
 
-BalanceTab.prototype.angular = function (module)
-{
-  
+BalanceTab.prototype.angular = function(module) {
 
-  module.controller('BalanceCtrl', ['$rootScope', 'rpId',
-                                     function ($scope, $id)
-  {
-    if (!$id.loginStatus) $id.goId();
-  }]);
+
+  module.controller('BalanceCtrl', ['$rootScope', 'rpId', 'rpNW',
+    function($scope, $id, rpNW) {
+      if (!$id.loginStatus) {
+        $id.goId();
+      } else {
+        rpNW.initTray();
```

Comment:
@vhpoet calling initTray again reinitializes the existing tray with new menu depending on the user's authorization status.